pilot/pkg/model/service.go

```diff
@@ -593,6 +630,16 @@ func (ep *IstioEndpoint) CmpOpts() []cmp.Option {
 	return istioEndpointCmpOpts
 }
 
+// Key returns the key of IstioEndpoint based on its addresses
+func (ep *IstioEndpoint) Key() string {
```

Comment:
I don't think we need to key on them all. Addresses[0] uniquely identifies the endpoint, no?

Reply:
Hi @howardjohn, I believe that `Addresses[0]` can be uniquely identifier for most scenarios, and in fact most of the Key should be `Addresses[0]`.  I can change to use the first address if so.